### PR TITLE
Implemented Bedrock Jurassic ChatClient

### PR DESCRIPTION
**Pull request by hemeda3**

### CODE REVIEW

**Comment by tzolov** [comment]
@hemeda3, great to see your contribution!

We'll try to review as a part of 1.0.0-M1.
In the mean time you can:
-  add the autoconfiguration class to the [org.springframework.boot.autoconfigure.AutoConfiguration.imports](https://github.com/spring-projects/spring-ai/blob/main/spring-ai-spring-boot-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports)
- create an mistral boot starter project under the: https://github.com/spring-projects/spring-ai/tree/main/spring-ai-spring-boot-starters
- add the required dependencies to the BOM: https://github.com/spring-projects/spring-ai/blob/main/spring-ai-bom/pom.xml
- and add documentation reference to the https://github.com/spring-projects/spring-ai/tree/main/spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/bedrock

(Feel free to ping me if you need help with any of those)

File: models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatOptions.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        this.temperature = temperature;
+    }
+
+    public Float getTopP() {
+        return topP;
+    }
+
+    public void setTopP(Float topP) {
+        this.topP = topP;
+    }@Override public Integer getTopK() {
```
Comment (laglangyue) [inline models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatOptions.java:64]: ` mvn spring-javaformat:apply  ` can format your code

**Comment by hemeda3** [comment]
@tzolov  Thanks for the feedback 🙌 working on it 

> * create an mistral boot starter project under the: https://github.com/spring-projects/spring-ai/tree/main/spring-ai-spring-boot-starters

I can see there is already a starter project for bedrock `spring-ai-bedrock-ai-spring-boot-starter` which I think includes `Ai21Jurassic2ChatBedrockApi`

So should we create new one and call it ( for example : `spring-ai-bedrock-**jurassic2/ai21**-ai-spring-boot-starter` ) ?

**Comment by tzolov** [comment]
@hemeda3 , 
ah, of course we don't. My bad. Forgot we collocate all bedrock auto-configurations into a single boot starter. 
You only need to make sure that the Jurassic auto-configuration is disabled by default using an enabled property like [this ](https://github.com/spring-projects/spring-ai/blob/3a59587357a219323186659d4b05d676c099bd6d/spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/bedrock/llama2/BedrockLlama2ChatProperties.java#L38).

**Comment by hemeda3** [comment]
When `spring.ai.bedrock.jurassic2.chat.enabled=false` or when the property is  absent ✅

image

When `spring.ai.bedrock.jurassic2.chat.enabled=true` ✅

image

**Comment by hemeda3** [comment]
**Should we change the name of `BedrockAi21Jurassic2ChatOptions` to `BedrockJurassic2ChatOptions` ?**

File: models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatOptions.java (hunk 1 above)
Comment (hemeda3) [reply models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatOptions.java]: Thanks  🙌 .
Fixed

File: models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatClient.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package org.springframework.ai.bedrock.jurassic2;
```
Comment (laglangyue) [inline models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatClient.java:1]: lost the license header, please check all files.

Comment (hemeda3) [reply models/spring-ai-bedrock/src/main/java/org/springframework/ai/bedrock/jurassic2/BedrockAi21Jurassic2ChatClient.java:17]: @laglangyue  Fixed for all.

**Comment by tzolov** [comment]
Hey @hemeda3 ,
Finally I've got time to review your PR. 
Good job. All checkbox checked including the documentation. 
I did some small adjustments before i can merge it and noticed that the `BedrockAi21Jurassic2ChatClientIT.emojiPenaltyWhenTrueByDefaultApplyPenaltyTest()` fails when run in combination with the other jurassic2 tests.  I've disabled it for the merge, but feel free to investigate .

**Comment by tzolov** [comment]
Rebased, squashed and merged at c036931065730619a9fb01c90fbe63d7ffbccb1d

**Pull request closed without merging**